main.py

```diff
@@ -0,0 +1,338 @@
+    response_model_exclude_unset=True,
+    response_model_exclude_defaults=True,
+    response_model_exclude_none=True,
+)
+def query_dataset_types_re(regex: Optional[List[str]] = Query(None),
+                           glob: Optional[List[str]] = Query(None),
+                           components: Optional[bool] = None) -> List[SerializedDatasetType]:
+    butler = Butler(butler=GLOBAL_BUTLER)
+    expression_params = ExpressionQueryParameter(regex=regex, glob=glob)
+
+    datasetTypes = butler.registry.queryDatasetTypes(expression_params.expression(),
+                                                     components=components)
+    return [d.to_simple() for d in datasetTypes]
+
+
+@subapp.get("/v1/registry/collectionChain/{parent:path}")
+def get_collection_chain(parent: str) -> CollectionSearch:
+    butler = Butler(butler=GLOBAL_BUTLER)
+    chain = butler.registry.getCollectionChain(parent)
+    return chain
+
+
+@subapp.get("/v1/registry/collections")
+def query_collections(regex: Optional[List[str]] = Query(None),
+                      glob: Optional[List[str]] = Query(None),
+                      datasetType: Optional[str] = None,
+                      flattenChains: Optional[bool] = False,
+                      collectionType: Optional[List[str]] = Query(None, regex=COLLECTION_TYPE_RE),
+                      includeChains: Optional[bool] = None) -> List[str]:
+
+    expression_params = ExpressionQueryParameter(regex=regex, glob=glob)
+
+    if collectionType is None:
+        collectionTypes = CollectionType.all()
+    else:
+        # Convert to real collection types
+        collectionTypes = set()
+        for item in collectionType:
+            item = item.upper()
+            try:
+                collectionTypes.add(CollectionType.__members__[item])
+            except KeyError:
+                raise KeyError(f"Collection type of {item} not known to Butler.")
+
+    butler = Butler(butler=GLOBAL_BUTLER)
+    collections = butler.registry.queryCollections(expression=expression_params.expression(),
+                                                   datasetType=datasetType,
+                                                   collectionTypes=collectionTypes,
+                                                   flattenChains=flattenChains,
+                                                   includeChains=includeChains)
+    return list(collections)
+
+
+@subapp.get("/v1/registry/collection/type/{name:path}")
+def get_collection_type(name: str):
+    butler = Butler(butler=GLOBAL_BUTLER)
+    collectionType = butler.registry.getCollectionType(name)
+    return collectionType.name
+
+
+@subapp.put("/v1/registry/collection/{name:path}/{type_}")
+def register_collection(name: str, type_: str, doc: Optional[str] = None) -> str:
+    type_ = type_.upper()
+    try:
+        collectionType = CollectionType.__members__[type_]
+    except KeyError:
+        raise KeyError(f"Collection type of {type_} not known to Butler.")
```

Comment:
This folds into the discussion in Jira on where the boundary is between knowing it's a fastapi server and it being standard python.

Reply:
Well, this is in the handler, so it is definitely and without question a FastAPI server. I think I see what you mean about asking what the client should do, but nothing on the client side will get easier if it gets a 500 reply instead of a 404, so I think you want the server to return 404 anyway.

Reply:
I changed this to use an enum and told FastAPI that it was that enum so I think that will help.